src/routing/annotations.cr

```diff
@@ -0,0 +1,70 @@
+module Athena::Routing
+  # :nodoc:
+  @@server : HTTP::Server?
+
+  # Defines a GET endpoint.
+  # ## Fields
+  # * path : `String` - The path for the endpoint.
+  #
+  # ## Example
+  # ```
+  # @[Athena::Routing::Get(path: "/users")]
+  # ```
+  annotation Get; end
+
```

Comment:
Should be a `PATCH` method added?

Reply:
Probably should also add `HEAD` and `OPTIONS` as well?  Reminds me I need to handle doing `HEAD` requests as well...